src/components/Relation.vue

```diff
@@ -339,235 +300,241 @@
-              layer:    this.layer,
-              fid:      this.feature.attributes[G3W_FID],
-              relation: this.relation,
-              page,
-              page_size,
-              ordering,
-            })
-          }); // get relations
-          features = response.result ? (response.vector.data.features || []).map(f => {
-            f.properties[G3W_FID] = f.id;
-            return {
-              geometry:   f.geometry,
-              attributes: f.properties,
-              id:         f.id,
+
+          GUI.setLoadingContent(true);
+
+          // Destroy previous table
+          if (this.$table) {
+            this.$table.destroy(true);
+            this.$table     = null;
+            this.table.rows = [];
+            await this.$nextTick();
+          }
+
+          try {
+
+            const changed = (opts.page_size !== prevOpts.page_size || opts.start !== prevOpts.start);
+
+            // merge options: (defaults + old + new) 
+            opts = {
+              /** @type { "desc" | "asc" | "current" } column order */
+              sort: 'asc',
+              /** @type { string | undefined } parameter sent to server ("-" = descending ) only if user has already clicked on a column to sort data */
+              ordering: undefined,
+              start: 0,
+              sort_column: 0,
+              order: [],
+              ...prevOpts,
+              ...opts
             };
-          }) : null;
-          const count = features && response.vector.count;
-          if (this.nmRelation) {
-            relationLayerId = this.nmRelation.referencedLayer;
-            features = await this.getRelationsNM({
-              nmRelation: this.nmRelation,
-              features
+
+
+            // check if ordering is asked by user by click on a column
+            if (opts.ordering && changed) {
+              opts.sort = opts.order[0].dir; // in case of change page, get last sorting of column
+            }
+
+            // in case of sort columns is not a previous column
+            if (opts.ordering && !changed && opts.order[0].column !== opts.sort_column) {
+              opts.sort = 'asc';
+            }
+
+            // invert sort
+            if (opts.ordering && !changed && opts.order[0].column === opts.sort_column) {
+              opts.sort = ('desc' === opts.sort ? 'asc' : 'desc');
+            }
+
+            // in case of set ordering (field sort) or not set ordering (start time)
+            if (opts.ordering || opts.order.length) {
+              opts.ordering = `${'desc' === opts.sort ? '-' : ''}${this.table.fields[opts.order[0].column - Number(!!this.showTools)].name}`;
+            }
+
+            /** @type { number } current column index */
+            opts.sort_column = opts.ordering && (this.table.fields.findIndex(({ name }) => ('desc' === opts.sort ? opts.ordering.slice(1) : opts.ordering) === name) + Number(!!this.showTools) ); //need to add 1 if threa are some 
```

Comment:
@volterra79 pleas double check, I think it can be simplified further (ref: `prevOpts`)

Reply:
@Raruto i'll check